ios/FluentUI.Demo/FluentUI.Demo/Demos/PillButtonBarDemoController.swift

```diff
@@ -202,3 +205,65 @@ extension PillButtonBarDemoController: PillButtonBarDelegate {
         present(alert, animated: true)
     }
 }
+
+extension PillButtonBarDemoController: DemoAppearanceDelegate {
+    func themeWideOverrideDidChange(isOverrideEnabled: Bool) {
+        guard let fluentTheme = self.view.window?.fluentTheme else {
+            return
+        }
+
+        var tokensClosure: ((PillButton) -> PillButtonTokens)?
+        if isOverrideEnabled {
+            tokensClosure = { _ in
+                return ThemeWideOverridePillButtonTokens()
+            }
+        }
+
+        fluentTheme.register(controlType: PillButton.self, tokens: tokensClosure)
+    }
+
+    func perControlOverrideDidChange(isOverrideEnabled: Bool) {
+        self.bars.forEach({ bar in
+            let tokens = isOverrideEnabled ? PerControlOverridePillButtonTokens() : nil
+            bar.pillButtonOverrideTokens = tokens
+        })
+    }
+
+    func isThemeWideOverrideApplied() -> Bool {
+        return self.view.window?.fluentTheme.tokenOverride(for: PillButton.self) != nil
+    }
+
+    // MARK: - Custom tokens
+
+    private class ThemeWideOverridePillButtonTokens: PillButtonTokens {
+        override var font: FontInfo {
+            return FontInfo(name: "Times", size: 15.0, weight: .regular)
+        }
+    }
+
+    private class PerControlOverridePillButtonTokens: PillButtonTokens {
+        override var backgroundColor: PillButtonDynamicColors {
+            return .init(rest: DynamicColor(light: ColorValue(r: CGFloat.random(in: 0.25...0.5),
```

Comment:
talked offline: random colors in demo app isn't advise. we try to make sure all colors in demo app is accessible to demonstrate good color contrast guidance.